docs/src/main/asciidoc/amazon-sns.adoc

```diff
@@ -0,0 +1,445 @@
+* Credentials delivered through the Amazon EC2 container service if the `AWS_CONTAINER_CREDENTIALS_RELATIVE_URI` environment variable is set and the security manager has permission to access the variable,
+* Instance profile credentials delivered through the Amazon EC2 metadata service
+
+== Next steps
+
+=== Packaging
+
+Packaging your application is as simple as `./mvnw clean package`.
+It can be run with `java -jar target/amazon-sns-quickstart-1.0-SNAPSHOT-runner.jar`.
+
+With GraalVM installed, you can also create a native executable binary: `./mvnw clean package -Dnative`.
+Depending on your system, that will take some time.
+
+=== Going asynchronous
+
+Thanks to the AWS SDK v2.x used by the Quarkus extension, you can use the asynchronous programming model out of the box.
+
+Create a `org.acme.sns.QuarksAsyncResource` REST resource that will be similar to our `QuarksSyncResource` but using an asynchronous programming model.
+
+[source,java]
+----
+package org.acme.sns;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import io.smallrye.mutiny.Uni;
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.acme.sns.model.Quark;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.jboss.logging.Logger;
+import software.amazon.awssdk.services.sns.SnsAsyncClient;
+import software.amazon.awssdk.services.sns.model.PublishResponse;
+
+@Path("/async/cannon")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class QuarksAsyncCannonResource {
+
+    private static final Logger LOGGER = Logger.getLogger(QuarksAsyncCannonResource.class);
+
+    @Inject
+    SnsAsyncClient sns;
+
+    @ConfigProperty(name = "topic.arn")
+    String topicArn;
+
+    static ObjectWriter QUARK_WRITER = new ObjectMapper().writerFor(Quark.class);
+
+    @POST
+    @Path("/shoot")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Uni<Response> publish(Quark quark) throws Exception {
+        String message = QUARK_WRITER.writeValueAsString(quark);
+        return Uni.createFrom()
+            .completionStage(sns.publish(p -> p.topicArn(topicArn).message(message)))
+            .onItem().invoke(item -> LOGGER.infov("[SYNC] Fired Quark[{0}, {1}}]", quark.getFlavor(), quark.getSpin()))
+            .onItem().apply(PublishResponse::messageId)
+            .onItem().apply(id -> Response.ok().entity(id).build());
+    }
+}
+----
+We create `Uni` instances from the `CompletionStage` objects returned by the asynchronous SNS client, and then transform the emitted item.
+
+And add Netty HTTP client dependency to the `pom.xml`:
+
+[source,xml]
+----
+<dependency>
+    <groupId>software.amazon.awssdk</groupId>
+    <artifactId>netty-nio-client</artifactId>
+</dependency>
+----
```

Comment:
Could we have that one first? It seems more level than the Java implementation?

Reply:
What do you mean by that ?